### PR TITLE
add precompile verification to simulate_transaction

### PR DESCRIPTION
**Pull request by oJshua**
#### Problem
While attempting to simulate a secp256k1 instruction transaction, it was discovered that an invalid secp256k1 instruction won't be handled due to simulate_transaction's lack of precompile verification.

#### Summary of Changes
Introduces precompile verification to simulate_transaction call.

Fixes https://github.com/solana-labs/solana/issues/13079

### CODE REVIEW

File: core/src/rpc.rs [hunk 1]
```diff
@@ -2322,6 +2322,10 @@ impl RpcSol for RpcSolImpl {
             Ok(())
         };
 
+        if let Err(e) = transaction.verify_precompiles() {
```
Comment (mvines) [inline core/src/rpc.rs:2325]: I think this should be under the `if config.sig_verify` condition above

Comment (oJshua) [reply core/src/rpc.rs]: Roger!

File: core/src/rpc.rs [hunk 2]
```diff
@@ -2317,7 +2317,11 @@ impl RpcSol for RpcSolImpl {
         let (_, transaction) = deserialize_transaction(data, encoding)?;
 
         let mut result = if config.sig_verify {
-            transaction.verify()
+            if let Err(e) = transaction.verify() {
```
Comment (mvines) [inline core/src/rpc.rs:2320]: Let's use the exact form as `send_transaction`:
```
            if transaction.verify().is_err() {
                return Err(RpcCustomError::TransactionSignatureVerificationFailure.into());
            }
            if let Err(e) = transaction.verify_precompiles() {
                return Err(RpcCustomError::TransactionPrecompileVerificationFailure(e).into());
            }
```
No need for possible return value differences between these two APIs.  Consider moving the verification into a shared function even

Comment (oJshua) [reply core/src/rpc.rs:2320]: Okay, good plan

**Review by mvines — APPROVED** [review]

**Pull request merged**